### ISSUE TITLE
Feature: Zeitspanne "Enddatum"

### ISSUE THREAD

**Issue by drimed** [iamnotturner/vaccipy#281]
Hi,

es wäre super, wenn im Menü "Zeitspanne" auch "Termin bis spätestens" angeboten werden könnte.

Anwendungsfall: Aufgrund einer Zeckenimpfung ist man bis 04.06.2021 für die Impfung gesperrt. Die zweite Zeckenimpfung ist Mitte Juli, ein Impftermin ist also nur bis zum 01.07.2021 möglich (14 Tage müssen zwischen den Impfungen liegen).

**Comment by andoorius** [comment]
Aktuell gibt es so wenig Termine, m.E. Risiko einen Termin im Juli zu bekommen = 0% leider. Meine Erfahrung in den letzten Tagen zw. Buchung und Termin oft nur Stunden bis Max. 5 Tage.
Falls es irgendwann mal genügend Impfstoff/Termine gibt bestimmt interessant (und falls es dann noch Impfzentren gibt)

**Comment by drimed** [comment]
Mir ist es aber passiert, das ich mit meinem Code einen Termin Mitte Juli buchen konnte. Jetzt wo Betriebe und Hausärzte mehr impfen können, schätze ich vermehrt Absagen bei späteren Terminen.

**Comment by Sars-Cov-2-Antigen** [comment]
Muss diese Impfpause nicht auch zwischen der Zweitimpfung und der Zeckenimpfung liegen?
Sofern Du nicht J&J als Impfstoff wählst, kannst Du die Termine immer nur paarweise buchen. Bei einer Erstimpfung in den kommenden Tagen, fiele deine Zweitimpung automatisch auf Mitte Juli. Du bräuchtest also einen Erstimpftermin zwischen dem 21.06 und dem 01.07.
So weit im Voraus habe ich bisher aber auch keinen Terminvorschlag erhalten.

**Comment by drimed** [comment]
Bei FSME hast du 1-3 Monate. Bei Biontech genau 6 Wochen.
Das ist also an sich kein Problem, die Abstände korrekt einzuhalten, sollte ein Termin kurzfristig zur Verfügung stehen.

Dies war ja auch nur ein möglicher Anwendungsfall, es gibt sicherlich noch andere Gründe warum ein Enddatum im Suchzeitraum Sinn ergibt.

**Comment by Sars-Cov-2-Antigen** [comment]
> 
> 
> Bei FSME hast du 1-3 Monate. Bei Biontech genau 6 Wochen.
> Das ist also an sich kein Problem, die Abstände korrekt einzuhalten, sollte ein Termin kurzfristig zur Verfügung stehen.
> 
> Dies war ja auch nur ein möglicher Anwendungsfall, es gibt sicherlich noch andere Gründe warum ein Enddatum im Suchzeitraum Sinn ergibt.

Grundsätzlich kann es nützlich sein. Ich bin nicht gegen deinen Vorschlag.

Ich hatte Deine Situation nur so verstanden, dass Du Mitte Juli eine Zeckenimpfung erhälst und deshalb 2 Wochen davor und danach (praktisch den gesamten Juli über) nicht eine weitere Impfung erhalten kannst. Meiner Meinung nach kannst Du problemlos nach Terminen ab dem 21.06 suchen. Du wirst aktuell wohl ohnehin keine bekommen und die Gefahr, dass Du im Juli einen erhälst, ist sehr gering. Wenn doch, kannst Du ihn immer noch absagen.

**Comment by funnym0nk3y** [comment]
Ich fände dieses Feature auch sehr nützlich. Ich kann in einem Zeitraum von 6 Wochen nur in zwei Wochen geimpft werden. Das würde dann so aussehen:

Ein Termin sperrt die Corona-Impfung für 2 Wochen. Der Termin muss frühestens nach 4 Wochen und spätestens nach 6 Wochen wiederholt werden und darf frühestens 2 Wochen nach der Corona-Impfung sein. Ergo gibt es wenige Tage im Monat die möglich sind.

**Comment by drimed** [comment]
Ich habe gerade auch erneut einen Termin im Juli bekommen, den ich auch nicht annehmen kann. Eine entsprechende Funktion wäre also super... ;-)

**Comment by andoorius** [comment]
Verrückt! Ich bekomme irgendwie immer Termine in wenigen Stunden/Tagen.

Suchst Du in BaWü?

**Comment by drimed** [comment]
Nein, in HH. Hier gibt es kaum Termine. Das muss auch ein Absagerslot gewesen sein...

**Comment by andoorius** [comment]
Ja ok. Habe in HH auch für Freunde was gebucht und 3 Wochen in der Zukunft.... ich frage mich, warum es in HH so viele Zentren gibt mit 60+ und die Webseite Dir auch nach Eingabe von einem Alter deutlich unter 60 einen Termin bucht....

**Comment by funnym0nk3y** [comment]
Ich habe gerade den Code angeschaut, und falls ich das richtig verstanden habe, müsste der das schon können. Zumindest in der Art. Zwischen den Terminen liegen 40 Tage. Damit kann man ausrechnen, wann der erste Termin sein muss, um mit dem zweiten was passendes zu bekommen.

**Comment by drimed** [comment]
In einem der letzten betas kam es auf Kommandozeile dazu. Lasse aktuell bis Ende Juni für den ersten Termin suchen. Ob das so klappt weiß ich noch nicht, mangels verfügbarer Termine.

**Comment by prok42** [comment]
@drimed Wenn der 2. Termin der Corona Impfung nicht passt, kann man den (mehrfach gelesen, nicht selbst getestet) über die Hotline verschieben lassen, sprich, man muss nicht das ganze Terminpaar absagen. Wäre vlt. eine Option.

**Comment by drimed** [comment]
Danke für deinen Tipp! Mit dem Tool hier und viel Glück habe ich nun meine Impfung morgen :-) Es hat sich also alles positiv entwickelt :-)
